From 249770e3f16af9848aeee6e182e293769c3c06c8 Mon Sep 17 00:00:00 2001
From: Sivasubramanaiyan Padmanaban <sivasubramaniyan.padmanaban@analog.com>
Date: Wed, 7 Jan 2026 14:36:04 +0530
Subject: [PATCH] drivers: media: i2c: adsd3500.c: enable FSYNC PWM only in PWM
 trigger mode

Switch FSYNC PWM enable/disable checks from FSYNC_HIZ to PWM_TRIGGER
so PWM is driven only when the sensor is configured for PWM sync.

Signed-off-by: Sivasubramanaiyan Padmanaban <sivasubramaniyan.padmanaban@analog.com>
---
 drivers/media/i2c/adsd3500.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/adsd3500.c b/drivers/media/i2c/adsd3500.c
index e0f105aebe61..821414818de0 100755
--- a/drivers/media/i2c/adsd3500.c
+++ b/drivers/media/i2c/adsd3500.c
@@ -1459,7 +1459,7 @@ static int adsd3500_start_streaming(struct adsd3500 *adsd3500)
 	if (ret < 0)
 		dev_err(adsd3500->dev, "Write of STREAM-ON command failed.\n");
 
-	if(adsd3500->pwm_fsync != NULL && adsd3500->curr_sync_mode == FSYNC_HIZ ){
+	if(adsd3500->pwm_fsync != NULL && adsd3500->curr_sync_mode == PWM_TRIGGER ){
 		ret = pwm_enable(adsd3500->pwm_fsync);
 		if (ret)
 			dev_err(adsd3500->dev, "Could not enable FSYNC PWM\n");
@@ -1475,7 +1475,7 @@ static int adsd3500_stop_streaming(struct adsd3500 *adsd3500)
 
 	dev_info(adsd3500->dev, "%s\n", __func__);
 
-	if(adsd3500->pwm_fsync != NULL && adsd3500->curr_sync_mode == FSYNC_HIZ){
+	if(adsd3500->pwm_fsync != NULL && adsd3500->curr_sync_mode == PWM_TRIGGER){
 		pwm_disable(adsd3500->pwm_fsync);
 		msleep(1000 / adsd3500->framerate);
 	}
-- 
2.25.1

